runtime/cpp/emboss_arithmetic.h

```diff
@@ -150,26 +175,68 @@ struct GreaterThanOrEqualOperation {
 // MaximumOperation is a bit more complex, in order to handle the variable
 // number of parameters.
 struct MaximumOperation {
+  // Maximum of 1 element is just itself.
   template <typename T>
   static inline constexpr T Do(T arg) {
-    // Base case for recursive template.
     return arg;
   }
 
-  // Ideally, this would only use template<typename T>, but C++11 requires a
-  // full variadic template or C-style variadic function in order to accept a
-  // variable number of arguments.  C-style variadic functions have no intrinsic
-  // way of figuring out how many arguments they receive, so we have to use a
-  // variadic template.
+  // The rest of MaximumOperation::Do could be:
+  //
+  // template <typename T, typename... RestT>
+  // static inline constexpr T Do(T v0, T v1, RestT... rest) {
+  //   return Do(v0 < v1 ? v1 : v0, rest...);
+  // }
   //
-  // The static_assert ensures that all arguments are actually the same type.
-  template <typename T1, typename T2, typename... T>
-  static inline constexpr T1 Do(T1 l, T2 r, T... rest) {
+  // ... unfortunately, some compilers do not optimize this well, and it ends
+  // up using linear stack space instead of constant stack space; for complex
+  // structs on systems with limited stack (such as typical microcontrollers),
+  // this can cause methods like Ok() to blow the stack.
+  //
+  // The C++14 solution would be to use a std::initializer_list and iterate over
+  // the arguments.  Unfortunately, C++11 std::initializer_list is not
+  // constexpr, and C++11 constexpr does not allow iteration.
+  //
+  // Instead, we have a small number of hand-written overloads and a large
+  // number (61, at time of writing) of generated overloads, which use O(lg(N))
```

Comment:
Perhaps:
`(61, at time of writing, controlled by OVERLOADS in maximum_operation_do.py)`
But I think it's 64 - 5 = 59, isn't it?

Reply:
Done.